.github/workflows/rust-ci.yaml

```
@@ -128,3 +128,55 @@ jobs:
           YARA_INCLUDE_DIR: ${{ github.workspace }}/.yara/${{ needs.yara-test.outputs.include-dir }}
           YARA_LIBRARY_PATH: ${{ github.workspace }}/.yara/${{ needs.yara-test.outputs.library-path }}
         run: cargo test --no-fail-fast
+
+  docs-build:
+    needs: [yara]
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+
+      - name: Cache dependencies
+        uses: Swatinem/rust-cache@dd05243424bd5c0e585e4b55eb2d7615cdd32f1f # v2.5.1
+        with:
+          key: x86_64-unknown-linux-gnu
+
+      - name: Download YARA
+        uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
+        with:
+          name: ${{ needs.yara.outputs.artifacts }}
+          path: .yara
+
+      - name: "Build docs with cargo"
+        run: cargo doc --no-deps --document-private-items
+        env:
+          RUST_BACKTRACE: full
+          YARA_INCLUDE_DIR: ${{ github.workspace }}/.yara/${{ needs.yara.outputs.include-dir }}
+          YARA_LIBRARY_PATH: ${{ github.workspace }}/.yara/${{ needs.yara.outputs.library-path }}
+
+      - name: "Upload artifact"
+        uses: actions/upload-pages-artifact@a753861a5debcf57bf8b404356158c8e1e33150c # v2.0.0
+        with:
+          path: ./target/doc
+
+  docs-deploy:
+    if: ${{ github.ref == 'refs/heads/main' }}
+
+    # Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+    permissions:
+      contents: read
+      pages: write
+      id-token: write
+
+    environment:
+      name: github-pages
```

To be clear about what's deployed on the environment.
```suggestion
      name: docs
```

---

Will this work? I thought that specific name was required for this to work.
We can give it a try.